crates/sui-source-validation/src/lib.rs

```diff
@@ -617,15 +617,37 @@ fn download_and_compile(
         // Check the platform and proceed if we can download a binary. If not, the user should follow error instructions to sideload the binary.
         let platform = detect_platform(&root, compiler_version, &dest_canonical_binary)?;
         // Download if binary does not exist.
-        let url = format!("https://github.com/MystenLabs/sui/releases/download/mainnet-v{}/sui-mainnet-v{}-{}.tgz", compiler_version, compiler_version, platform);
+        let mainnet_url = format!(
+            "https://github.com/MystenLabs/sui/releases/download/mainnet-v{}/sui-mainnet-v{}-{}.tgz",
+            compiler_version, compiler_version, platform
+        );
 
         println!(
-            "{} compiler @ {} (this may take a while)",
+            "{} mainnet compiler @ {} (this may take a while)",
             "DOWNLOADING".bold().green(),
             compiler_version.yellow()
         );
 
-        let mut response = ureq::get(&url).call()?.into_reader();
+        let mut response = match ureq::get(&mainnet_url).call() {
+	    Ok(response) => response,
+	    Err(ureq::Error::Status(404, _)) => {
```

Comment:
nit: There is something funky going on with whitespace here -- perhaps mixing tabs and spaces?

Reply:
Cheers, it turns out the rust formatter was choking on the `println` macros. I've fixed it manually